lib/util/fake_event_target.js

```diff
@@ -104,19 +104,50 @@ shaka.util.FakeEventTarget.prototype.removeEventListener =
  * @override
  */
 shaka.util.FakeEventTarget.prototype.dispatchEvent = function(event) {
-  // Overwrite the Event's properties.  Assignment doesn't work in most
-  // browsers.  Object.defineProperties seems to work, although some browsers
+  // Overwrite the Event's properties if not already done so (events can be
+  // re-dispatched, so we may be have already done this, so in this case we
+  // can just update the values). Assignment doesn't work in most browsers.
+  // Object.defineProperty seems to work, although some browsers
   // need the original properties deleted first.
-  delete event.srcElement;
-  delete event.target;
-  delete event.currentTarget;
-  Object.defineProperties(event, {
-    'srcElement': { 'value': null, 'writable': true },
-    // target may be set many times if an event is re-dispatched.
-    'target': { 'value': this, 'writable': true },
-    // currentTarget will be set many times by recursiveDispatch_().
-    'currentTarget': { 'value': null, 'writable': true }
-  });
+
+  if (event.hasOwnProperty('srcElement')) {
+    event.srcElement = null;
+  }
+  else {
+    delete event.srcElement;
+
+    var srcElement_ = null;
+    Object.defineProperty(event, 'srcElement', {
+      get: function() { return srcElement_; },
```

Comment:
The local var is not needed, since srcElement is never set for fake events.  You can just return null, and kill the local and the setter.